metricflow/engine/metricflow_engine.py

```diff
@@ -320,13 +320,20 @@ def __init__(
         sql_client: SqlClient,
         time_source: TimeSource = ServerTimeSource(),
         column_association_resolver: Optional[ColumnAssociationResolver] = None,
+        enable_default_time_constraint: bool = True,
```

Comment:
Will this be expected for customers if we default to an adjusted time constraint? They might expect all time instead, in which case we should default to False. Definitely something we should make sure the docs reflect correctly.

Reply:
Actually, after looking at the old code, it seems like the adjustment feature was broken from the start, and it doesn't work correctly with derived metrics. I'm thinking that we remove it and add it as a query validation instead. Let me handle that in a follow up PR though.